src/main/scala/com/moesif/filter/MoesifApiFilter.scala

```diff
@@ -38,21 +38,34 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
   private val eventModelBuffer = mutable.ArrayBuffer[EventModel]()
   private val client = new MoesifAPIClient(moesifApplicationId, moesifCollectorEndpoint)
   private val moesifApi = client.getAPI
+  // App Config and Governance Rule refresh frequency in seconds
+  private val RefreshInterval = 300
+  // App Config load and refresh tracking
   private var lastTimeAppConfigFetched: DateTime = DateTime.parse("1970-01-01", DateTimeFormat.forPattern("yyyy-MM-dd"))
   private var isAppConfigFetched: Boolean = false
   private var appConfigModel = new AppConfigModel
+  // Governance Rules, refresh schedule managed by ScheduledExecutorService
+  private var governanceRules: Seq[GovernanceRulesModel] = Seq()
 
   val appConfigRunnable: Runnable = new Runnable() {
     override def run(): Unit = {
       getApplicationConfig()
     }
   }
+
+  val governanceRunnable: Runnable = new Runnable() {
+    override def run(): Unit = {
+      getGovernanceRules()
```

Comment:
Generally I add check/skip if `isJobRunning` to avoid spawning multiple threads.

> if (!isJobRunning) {
>    isJobRunning = true
>    // do stuffs
>    ......
>    isJobRunning = false
> }

Reply:
Great point! There were 3 slightly different ways of running recurring scheduled  code, so I went looking for a reliable way to standardize them, and it turns out that the scheduled executor service serializes executions of individual runnable, "subsequently with the given delay between the termination of one execution and the commencement of the next."